roles/istio/tasks/certificates.yaml

```diff
@@ -0,0 +1,144 @@
+################### Start Create CA Certificate ###################
+
+- name: Generate CA Private Key with RSA 2048
+  community.crypto.openssl_privatekey:
+    path: "{{ tmp_cert_dir.path }}/ca.key"
+    size: 2048
+
+
+- name: Generate CA CSR
+  community.crypto.openssl_csr:
+    path: "{{ tmp_cert_dir.path }}/ca.csr"
+    privatekey_path: "{{ tmp_cert_dir.path }}/ca.key"
+    common_name: "{{ istio_external_ip }}"
+    basic_constraints:
+      - "CA:TRUE"
+    basic_constraints_critical: true
+
+
+- name: Generate CA Self Signed Certificate
+  community.crypto.x509_certificate:
+    path: "{{ tmp_cert_dir.path }}/ca.crt"
+    privatekey_path: "{{ tmp_cert_dir.path }}/ca.key"
+    csr_path: "{{ tmp_cert_dir.path }}/ca.csr"
+    provider: selfsigned
+
+###################### End Create CA Certificate ######################
+################### Start Create Server Certificate ###################
+
+- name: Generate Server OpenSSL private key with RSA 2048
+  community.crypto.openssl_privatekey:
+    path: "{{ tmp_cert_dir.path }}/server.key"
+    size: 2048
+
+
+- name: Generate an OpenSSL Certificate Signing Request for server
+  community.crypto.openssl_csr:
+    path: "{{ tmp_cert_dir.path }}/server.csr"
+    privatekey_path: "{{ tmp_cert_dir.path }}/server.key"
+    common_name: "{{ istio_external_ip }}"
+    basic_constraints:
+      - "CA:FALSE"
+    key_usage:
+      - nonRepudiation
+      - digitalSignature
+      - keyEncipherment
+    extended_key_usage:
+      - serverAuth
+    subject_alt_name:
+      - "DNS:istio-ingressgateway" # TODO: put istio-ingressgateway and istio-system into variables? {{service}} {{namespace}}
+      - "DNS:istio-ingressgateway.istio-system"
+      - "DNS:istio-ingressgateway.istio-system.svc"
+      - "DNS:istio-ingressgateway.istio-system.svc.cluster"
+      - "DNS:istio-ingressgateway.istio-system.svc.cluster.local"
+
+
+# TODO: Original command generates a ca.srl file that is from the CAcreateserial flag.
+#       This is not generated with these steps at the moment but may not be required
+- name: Generate an OpenSSL certificate from the server CSR signed with our CA certificate
+  community.crypto.x509_certificate:
+    path: "{{ tmp_cert_dir.path }}/server.crt"
+    csr_path: "{{ tmp_cert_dir.path }}/server.csr"
+    ownca_path: "{{ tmp_cert_dir.path }}/ca.crt"
+    ownca_privatekey_path: "{{ tmp_cert_dir.path }}/ca.key"
+    provider: ownca
+    ## TODO: Figure out if these are necessary if already specified in the CSR?
+    key_usage:
+      - nonRepudiation
+      - digitalSignature
+      - keyEncipherment
+    extended_key_usage:
+      - serverAuth
+    subject_alt_name:
+      - "DNS:istio-ingressgateway"
+      - "DNS:istio-ingressgateway.istio-system"
+      - "DNS:istio-ingressgateway.istio-system.svc"
+      - "DNS:istio-ingressgateway.istio-system.svc.cluster"
+      - "DNS:istio-ingressgateway.istio-system.svc.cluster.local"
```

Comment:
I realise these values in the `x509_certificate` generation are used for **assertions**, which might be useful to keep

Reply:
Note that 
> The assertonly provider is intended for use cases where one is only interested in checking properties of a supplied certificate. Please note that this provider has been deprecated in Ansible 2.9 and will be removed in community.crypto 2.0.0. See the examples on how to emulate assertonly usage with community.crypto.x509_certificate_info, community.crypto.openssl_csr_info, community.crypto.openssl_privatekey_info and ansible.builtin.assert. This also allows more flexible checks than the ones offered by the assertonly provider.

Eventually, I will want to follow the example here:
```yaml
# The following example shows one assertonly usage using all existing options for
# assertonly, and shows how to emulate the behavior with the x509_certificate_info,
# openssl_csr_info, openssl_privatekey_info and assert modules:
- name: Usage of assertonly with all existing options
  community.crypto.x509_certificate:
    provider: assertonly
    path: /etc/ssl/crt/ansible.com.crt
    csr_path: /etc/ssl/csr/ansible.com.csr
    privatekey_path: /etc/ssl/csr/ansible.com.key
    signature_algorithms:
      - sha256WithRSAEncryption
      - sha512WithRSAEncryption
    subject:
      commonName: ansible.com
    subject_strict: yes
    issuer:
      commonName: ansible.com
    issuer_strict: yes
    has_expired: no
    version: 3
    key_usage:
      - Data Encipherment
    key_usage_strict: yes
    extended_key_usage:
      - DVCS
    extended_key_usage_strict: yes
    subject_alt_name:
      - dns:ansible.com
    subject_alt_name_strict: yes
    not_before: 20190331202428Z
    not_after: 20190413202428Z
    valid_at: "+1d10h"
    invalid_at: 20200331202428Z
    valid_in: 10  # in ten seconds

- name: Get certificate information
  community.crypto.x509_certificate_info:
    path: /etc/ssl/crt/ansible.com.crt
    # for valid_at, invalid_at and valid_in
    valid_at:
      one_day_ten_hours: "+1d10h"
      fixed_timestamp: 20200331202428Z
      ten_seconds: "+10"
  register: result

- name: Get CSR information
  community.crypto.openssl_csr_info:
    # Verifies that the CSR signature is valid; module will fail if not
    path: /etc/ssl/csr/ansible.com.csr
  register: result_csr

- name: Get private key information
  community.crypto.openssl_privatekey_info:
    path: /etc/ssl/csr/ansible.com.key
  register: result_privatekey

- assert:
    that:
      # When private key is specified for assertonly, this will be checked:
      - result.public_key == result_privatekey.public_key
      # When CSR is specified for assertonly, this will be checked:
      - result.public_key == result_csr.public_key
      - result.subject_ordered == result_csr.subject_ordered
      - result.extensions_by_oid == result_csr.extensions_by_oid
      # signature_algorithms check
      - "result.signature_algorithm == 'sha256WithRSAEncryption' or result.signature_algorithm == 'sha512WithRSAEncryption'"
      # subject and subject_strict
      - "result.subject.commonName == 'ansible.com'"
      - "result.subject | length == 1"  # the number must be the number of entries you check for
      # issuer and issuer_strict
      - "result.issuer.commonName == 'ansible.com'"
      - "result.issuer | length == 1"  # the number must be the number of entries you check for
      # has_expired
      - not result.expired
      # version
      - result.version == 3
      # key_usage and key_usage_strict
      - "'Data Encipherment' in result.key_usage"
      - "result.key_usage | length == 1"  # the number must be the number of entries you check for
      # extended_key_usage and extended_key_usage_strict
      - "'DVCS' in result.extended_key_usage"
      - "result.extended_key_usage | length == 1"  # the number must be the number of entries you check for
      # subject_alt_name and subject_alt_name_strict
      - "'dns:ansible.com' in result.subject_alt_name"
      - "result.subject_alt_name | length == 1"  # the number must be the number of entries you check for
      # not_before and not_after
      - "result.not_before == '20190331202428Z'"
      - "result.not_after == '20190413202428Z'"
      # valid_at, invalid_at and valid_in
      - "result.valid_at.one_day_ten_hours"  # for valid_at
      - "not result.valid_at.fixed_timestamp"  # for invalid_at
      - "result.valid_at.ten_seconds"  # for valid_in
```

Reply:
Nice, good to know but I don't think we need to this extend for now. Basic self-signed certs are good for most cases